libr/core/cbin.c

```diff
@@ -2391,6 +2433,25 @@ static int bin_classes(RCore *r, int mode) {
 				}
 			}
 			r_cons_printf ("]}");
+		} else if (IS_MODE_CLASSDUMP (mode)) { 
+      char *rp = NULL;
+      if (c) {
+        //TODO -> Print Superclass
+        r_cons_printf("@interface %s :  \n{\n", c->name); 
+        r_list_foreach (c->fields, iter2, f) {
+            if (f->name && r_regex_match("ivar","e", f->name)) {
+              r_cons_printf ("\t%s %s\n", f->type, f->name);
+            }
+        }
+        r_cons_printf("}\n");
+        r_list_foreach (c->methods, iter3, sym) {
+          if (sym->rtype && !r_str_cmp(sym->rtype,"@", 1)) {
+            rp = get_rp(sym->rtype);
+            r_cons_printf ("%s (%s) %s\n", r_str_cmp(sym->type,"METH",4) ? "-": "+", rp, sym->dname? sym->dname: sym->name);
```

Comment:
use strcmp, sym->type shoulkdnt be null and compilers know how to optimize that, r_str_cmp cant be optimized